pkgs/games/classicube/default.nix

```diff
@@ -0,0 +1,93 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+  # native build dependencies
+, dos2unix
+, makeWrapper
+  # build dependencies
+, SDL2
+, libGL
+, curl
+, openal
+  # runtime dependencies
+, liberation_ttf
+}:
+
+stdenv.mkDerivation rec {
+  pname = "ClassiCube";
+  version = "1.3.2";
+
+  src = fetchFromGitHub {
+    owner = "UnknownShadow200";
+    repo = "ClassiCube";
+    rev = version;
+    sha256 = "6a0f7b03ef3a7f74cf42ffa5b88ab1a7b7beb4d864871a1b700465343ae74bb6";
+  };
+
+  nativeBuildInputs = [ dos2unix makeWrapper ];
+
+  prePatch = ''
+    # The ClassiCube sources have DOS-style newlines
+    # which causes problems with diff/patch.
+    dos2unix 'src/Platform_Posix.c' 'src/Core.h'
+  '';
+
+  patches = [
+    # Fix hardcoded font paths
+    ./font-location.patch
+    # ClassiCube doesn't compile with its X11 backend
+    # because of issues with libXi.
+    ./use-sdl.patch
+    # For some reason, the Makefile doesn't link
+    # with libcurl and openal when ClassiCube requires them.
+    # Also links with SDL2 instead of libX11 and libXi.
+    ./fix-linking.patch
+  ];
+
+  font_path = "${liberation_ttf}/share/fonts/truetype";
+
+  enableParallelBuilding = true;
+
+  postPatch = ''
+    # ClassiCube hardcodes locations of fonts.
+    # This changes the hardcoded location
+    # to the path of liberation_ttf instead
+    substituteInPlace src/Platform_Posix.c \
+      --replace '%NIXPKGS_FONT_PATH%' "${font_path}"
+  '' + lib.optionalString enableParallelBuilding ''
+    # ClassiCube's Makefile hardcodes JOBS=1 for some reason,
+    # even though it works perfectly well multi-threaded.
+    substituteInPlace src/Makefile \
+      --replace 'JOBS=1' "JOBS=$NIX_BUILD_CORES"
+  '';
+
+  buildInputs = [ SDL2 libGL curl openal liberation_ttf ];
+
+  buildPhase = ''
+    cd 'src'
+    make
+    cd -
+  '';
```

Comment:
Please change the directory in preBuild.

Reply:
Done!